### ISSUE TITLE
Barcode scanner does not work

### ISSUE THREAD

**Issue by adipascu** [expo/expo#2949]
### Environment
I can reproduce this issue in the expo Android Standalone build, the Expo Client, and ExpoKit, all with SDK 31

### Steps to Reproduce

I can rarely scan barcodes using the Expo client, the sample code from the [expo documentation](https://docs.expo.io/versions/v31.0.0/sdk/bar-code-scanner#usage) or the [example snack](https://snack.expo.io/Skxzn6-5b).

Simplest way to reproduce the issue:
1. Install the Expo Client from the [Play Store](https://play.google.com/store/apps/details?id=host.exp.exponent)
2. Open and attempt to scan any barcode from the PC screen
3. Barcode does not scan

I debuged the native code to check what's up.
Tried running GMVBarCodeScanner and ZxingBarCodeScanner, both failed to scan.
I dumped a few NV21 frames from ram and decoded them locally with FFMPEG, they are not corrupted.
I tried scanning the dumped frames from my PC using zbarimg, and it also failed.
The resolution of the frames is 3840x2160.
[Here is one of the frames](https://user-images.githubusercontent.com/1521321/49736354-8e711300-fc92-11e8-93bc-4f03db45664a.jpg)

I think the resolution is too big and is causing the qr library to fail to recognize the QR codes correctly, also I think this causes a low scan speed in general.

**Comment by Szymon20000** [comment]
Could you give more informations? 
- Android version
- model of mobile phone

**Comment by adipascu** [comment]
Android 7.1.2 ([with this custom ROM](https://forum.xda-developers.com/oneplus-3/oneplus-3--3t-cross-device-development/rom-sultans-lineage-os-14-1-mar-23rd-t3767671)) on Oneplus 3T

**Comment by adipascu** [comment]
The same code works correctly on a Huawei p20 PRO device, the capture resolution is 1456x1456.
The Oneplus 3T woks correctly if I use the front camera, resolution 3840x2160 but the picture can not spot individual LCD pixels on my screen.
[Oneplus 3T selfie camera frame](https://user-images.githubusercontent.com/1521321/49750249-0ea77080-fcb3-11e8-862d-d8272156fbf6.png)

**Comment by Szymon20000** [comment]
ExpoBarCodeScannerView is passing preview to  ExpoBarCodeScanner (GMVBarCodeScanner or ZxingBarCodeScanner). Preview shoudn't be bigger then screen. Is it possible that the custom ROM has some custom internal camera implementation which may prevent BarCodeScanner from working properly?. Maybe camera in your system is returning only preview with poor resolution. You can check dimensions of preview by adding breakpoint in this [line](https://github.com/expo/expo/blob/6202ff63ce2b985d6c102aa445d746dcd75a413b/packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/ExpoBarCodeScanner.java#L133)

**Comment by adipascu** [comment]
Here are some screenshots ran on the Oneplus 3T
- [parameters.getSupportedPreviewSizes(), first entries](https://user-images.githubusercontent.com/1521321/49811390-d9128e00-fd6b-11e8-8ad1-16d03bd19213.png)
- [parameters.getSupportedPreviewSizes(), last entires](https://user-images.githubusercontent.com/1521321/49811388-d9128e00-fd6b-11e8-9f6d-a157f6388159.png)
- [optimalPreviewSize](https://user-images.githubusercontent.com/1521321/49811386-d9128e00-fd6b-11e8-85a1-f240bad1e24f.png)

**Comment by otaviogaiao** [comment]
I was having problems with the BarcodeScanner too, on Android. It would never call the onBarCodeScanned function. When I moved to the Camera component, then it works

**Comment by daxaxelrod** [comment]
I'm having very similar issues. Some days it doesn't pick up on anything, some days its perfect. This is across multiple devices including a genymotion vm using the webcam. I use the same QR code printout every time.

**Comment by CajetanRodrigues** [comment]
I just started building react native apps and the experience is so bad on expo
Not it just does not react to my QR code only, even though I scan it. react team plz address it

**Comment by 592da** [comment]
SDK 32 also has this issue :(

**Comment by daxaxelrod** [comment]
Just as a follow up, I ended up ejecting from the expo environment. It was tricky at first but once you figure out your gradle file, it becomes easier.

I now use https://github.com/react-native-community/react-native-camera for barcode scanning and have never once had an issue

**Comment by brentvatne** [comment]
@daxaxelrod - react-native-camera is expo-camera copy+pasted outside of expo

**Comment by brentvatne** [comment]
for folks that are running into this, please provide a reproducible example and info about exactly what device / os version you're using

**Comment by NicTorgersen** [comment]
Not working on Nokia 8.1, Android 9.

**Comment by devonwesley** [comment]
Did you fix you problem? @NicTorgersen

**Comment by NicTorgersen** [comment]
@devonwesley 
No, sadly I did not.

**Comment by devonwesley** [comment]
did you find a work around?

**Comment by NicTorgersen** [comment]
@devonwesley I doubt there is one, I **_suspect_** the camera hw for the Nokia 8.1 differ too much from other phones which makes it difficult to parse the photo. Disclaimer: this is speculation.

**Comment by devonwesley** [comment]
this ended up working for me.

```
        <Camera
          onBarCodeScanned={this.handleBarCodeScanned}
          style={[StyleSheet.absoluteFill, styles.qrContainer]}
          barCodeScannerSettings={{barCodeTypes: [BarCodeScanner.Constants.BarCodeType.qr]}}
        >
          <Text style={styles.description}>Scan your QR code</Text>
          <Image
            style={styles.qr}
            source={require('../assets/images/auto_focus-512.png')}
          />
          <Text
            onPress={() => this.setState({scanned: false})}
            style={styles.cancel}>
            Cancel
          </Text>
        </Camera>
```

https://github.com/expo/expo/issues/5212#issuecomment-525365307

**Comment by devonwesley** [comment]
import `Camera` instead of `BarCodeScanner`.

**Comment by NicTorgersen** [comment]
My problem is with launching my app with Expo. It seems like you're way past that problem. :)

**Comment by adipascu** [comment]
What expo versions are you using? This should be fixed in SDK34 in #2957 .
